cmd/cmd.go

```diff
@@ -78,6 +92,22 @@ func ExecInCoreSnap() {
 		}
 	}
 
+	// ensure we do not re-exec into an older version of snapd
+	currentSnapd, err := os.Stat(exe)
+	if err != nil {
+		logger.Noticef("cannot stat %q: %s", exe, err)
+		return
+	}
+	coreSnapSnapd, err := os.Stat(full)
+	if err != nil {
+		logger.Noticef("cannot stat %q: %s", full, err)
+		return
+	}
+	if currentSnapd.ModTime().After(coreSnapSnapd.ModTime()) {
```

Comment:
That means it will arbitrarily move back and forth depending on who touches last, completely independent of the content. Feels very dangerous.. easy to brick. Revision seems much more stable and reliable. Do I miss your intention?